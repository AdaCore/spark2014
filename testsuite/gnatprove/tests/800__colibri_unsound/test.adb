package body Test
is

   function Ident (M : Univ_Int) return Univ_Int
     is
   begin
      pragma Unreferenced (M);
      return 0;
   end Ident;

   procedure Test
     (MI  : Univ_Int)
   is
   begin
      pragma Assert (Ident (0) mod MI = 0 mod MI); --@ASSERT:FAIL
   end Test;

end Test;
