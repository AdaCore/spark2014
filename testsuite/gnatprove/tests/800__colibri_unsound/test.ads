with SPARK.Big_Integers;

package Test
with
  Elaborate_Body,
  Ghost
is

   type Word32 is mod 2 ** 32;

   subtype Univ_Int is SPARK.Big_Integers.Big_Integer;
   use type Univ_Int;

   package Word32_Conversion is
     new SPARK.Big_Integers.Unsigned_Conversions (Int => Word32);

   function Int_Of (W : Word32) return Univ_Int
     renames Word32_Conversion.To_Big_Integer;

   function Ident (M : Univ_Int) return Univ_Int;

   procedure Test
     (MI  : Univ_Int)
   with
     Pre => MI /= 0,
     Post => False;

end Test;
