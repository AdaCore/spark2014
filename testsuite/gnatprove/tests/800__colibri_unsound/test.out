test.adb:15:22: info: assertion proved (colibri: 1 VC)
test.adb:15:22: info: predicate check proved (Trivial: 1 VC)
test.adb:15:32: info: division check proved (CVC5: 1 VC)
test.adb:15:36: info: predicate check proved (Trivial: 1 VC)
test.adb:15:41: info: predicate check proved (Trivial: 1 VC)
test.adb:15:43: info: division check proved (CVC5: 1 VC)
test.adb:15:47: info: predicate check proved (Trivial: 1 VC)
test.ads:20:13: info: implicit aspect Always_Terminates on "Ident" has been proved, subprogram will terminate
test.ads:25:13: info: predicate check proved (Trivial: 1 VC)
test.ads:25:19: info: predicate check proved (Trivial: 1 VC)
test.ads:26:14: info: postcondition proved (colibri: 1 VC)
